ci/ray_ci/automation/test_ray_wheels_lib.py

```diff
@@ -0,0 +1,207 @@
+from unittest import mock
+import sys
+import tempfile
+import os
+from botocore.exceptions import ClientError
+import pytest
+
+from ci.ray_ci.automation.ray_wheels_lib import (
+    _get_wheel_names,
+    download_wheel_from_s3,
+    download_ray_wheels_from_s3,
+    _check_downloaded_wheels,
+    PYTHON_VERSIONS,
+    PLATFORMS,
+    RAY_TYPES,
+)
+
+SAMPLE_WHEELS = [
+    "ray-1.0.0-cp38-cp38-manylinux2014_x86_64",
+    "ray-1.0.0-cp38-cp38-manylinux2014_aarch64",
+    "ray-1.0.0-cp38-cp38-macosx_10_15_x86_64",
+    "ray-1.0.0-cp38-cp38-macosx_11_0_arm64",
+    "ray-1.0.0-cp38-cp38-win_amd64",
+]
+
+
+def test_get_wheel_names():
+    ray_version = "1.0.0"
+    wheel_names = _get_wheel_names(ray_version)
+
+    assert len(wheel_names) == len(PYTHON_VERSIONS) * len(PLATFORMS) * len(RAY_TYPES)
+
+    for wheel_name in wheel_names:
+        assert len(wheel_name.split("-")) == 5
+        (
+            ray_type,
+            ray_version,
+            python_version,
+            python_version2,
+            platform,
+        ) = wheel_name.split("-")
+        platform = platform.split(".")[0]  # Remove the .whl suffix
+
+        assert ray_type in RAY_TYPES
+        assert ray_version == ray_version
+        assert f"{python_version}-{python_version2}" in PYTHON_VERSIONS
+        assert platform in PLATFORMS
+
+
+def test_check_downloaded_wheels():
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        wheels = [
+            "ray-1.0.0-cp38-cp38-manylinux2014_x86_64",
+            "ray-1.0.0-cp38-cp38-manylinux2014_aarch64",
+            "ray-1.0.0-cp38-cp38-macosx_10_15_x86_64",
+            "ray-1.0.0-cp38-cp38-macosx_11_0_arm64",
+            "ray-1.0.0-cp38-cp38-win_amd64",
+        ]
+
+        for wheel in wheels:
+            with open(os.path.join(tmp_dir, wheel + ".whl"), "w") as f:
+                f.write("")
+
+        _check_downloaded_wheels(tmp_dir, wheels)
+
+
+def test_check_downloaded_wheels_fail():
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        wheels = [
+            "ray-1.0.0-cp38-cp38-manylinux2014_x86_64",
+            "ray-1.0.0-cp38-cp38-manylinux2014_aarch64",
+            "ray-1.0.0-cp38-cp38-macosx_10_15_x86_64",
+            "ray-1.0.0-cp38-cp38-macosx_11_0_arm64",
+            "ray-1.0.0-cp38-cp38-win_amd64",
+        ]
+
+        for wheel in wheels[:3]:
+            with open(os.path.join(tmp_dir, wheel + ".whl"), "w") as f:
+                f.write("")
+        with pytest.raises(Exception, match=" was not downloaded"):
```

Comment:
don't rely too much on the exception message for business logic

Reply:
fixed